go-controller/pkg/ovn/master_test.go

```diff
@@ -1678,6 +1679,103 @@ var _ = ginkgo.Describe("Default network controller operations", func() {
 		})
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 	})
+
+	ginkgo.It("verify IPs in host-network-namespace address_set after remote zone node add or delete", func() {
+		app.Action = func(ctx *cli.Context) error {
+			_, err := config.InitConfig(ctx, nil, nil)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			config.OVNKubernetesFeature.EnableInterconnect = true
+			config.Kubernetes.HostNetworkNamespace = "ovn-host-network"
+
+			// PortBinding is required to add interconect resources for
+			// remote zone nodes
+			pb := &sbdb.PortBinding{
+				LogicalPort: "tstor-node1",
+				Datapath:    "tstor-node1",
+				TunnelKey:   5,
+			}
+			dbSetup.SBData = append(
+				dbSetup.SBData,
+				pb,
+			)
+			// transit_switch is required to be able to start OVN with an
+			// empty NodeList
+			ts := &nbdb.LogicalSwitch{
+				Name: "transit_switch",
+			}
+			dbSetup.NBData = append(
+				dbSetup.NBData,
+				ts,
+			)
+
+			// https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/namespace.go#L312
+			// adds management port and gateway router LRP IP to HostNetworkNamespace address set whenever
+			// the namespace gets created. Hence starting OVN with an empty NodeList and adding the node
+			// after the namespace is created successfully
+			fakeOvn.startWithDBSetup(dbSetup,
+				&v1.NamespaceList{
+					Items: []v1.Namespace{*newNamespace(config.Kubernetes.HostNetworkNamespace)},
+				},
+				&v1.NodeList{
+					Items: []v1.Node{},
+				},
+			)
+
+			// Required to make sure a remote zone node is being added
+			fakeOvn.controller.zone = "node1"
+
+			gomega.Expect(fakeOvn.controller.WatchNamespaces()).To(gomega.Succeed(), "Namespace should be created fine")
+			err = fakeOvn.controller.WatchNodes()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
```

Comment:
add a check here to assert that the address set is empty

Reply:
Ack